### ISSUE TITLE
Allow non-BambuLab printers to have custom sheet profiles

### ISSUE THREAD

**Issue by HakunMatat4** [SoftFever/OrcaSlicer#2120]
**The problem**
**TLDR: Sheets have different offset and require to have their own profile forcing us to have many printers profile for each sheet..**

At the moment, only BambuLab printers have sheet profiles
![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/aaf1b8ef-40b3-4381-b9c9-658f6f5f0bbb)

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/c81cfdfa-1b9e-48cd-9f94-5f9994aaf59e)

But other printers don't use those sticker sheets that requires different temp, but we do need sheet profiles to set the offset.

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/9df00b90-30ee-4229-a097-d087af424817)

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/1606091e-6ec8-4ec4-ac3a-eedc8e6fe584)

-----
**Which printers will be beneficial to this feature**
I believe that all printers should benefit from this. 
BambuLab printers already have an offset block within start g-code.
Klipper definitely needs this, not sure about Marlin tho.
Prusa manages sheet offset within the printer firmware.

-----
**Describe the solution you'd like**
**SIMPLE**

Have sheet profiles enabled so we can update our MACROS to use the var `curr_bed_type`
Not just that but allow us to add sheet profiles just like we do with printer and filament.

Having High Temp Plate profile for example will make no sense at all but be able to allow us to custom the field with like Textured, Smooth, PEO, PEY, any other name.
This will remove any complexity from you guys and give us the freedom to create a sheet with the name that work for us.

-----
**Describe alternatives you've considered**
**ADVANCED**

On my Voron, with 5 different sheets with their unique offsets, I have this start g-code.

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/7039b774-3874-4e24-8d5d-1cddfb1e50e2)

On kllipper `PRINT_START` I have it set like so:

```
{% set z_offset = params.ZOFFSET|default(0.000)|float %}

{% if z_offset != 0 %}
     SET_GCODE_OFFSET Z_ADJUST={ z_offset } MOVE=1; SET Z-OFFSET
    {% endif %}
```
and `PRINT_STOP` to clean the off-set in case I change the sheet for the next print, otherwise, klipper will add any new offset on top of the current offset and poop will go down. 
Ask me how I found that out the hard way

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/ae5305ec-a52b-421e-b974-9906336b350f)

So the alternative would be:

1. have the sheet profiles enabled
2. add a sheet off-set field
3. **IF OFF-SET FIELD IS ENABLED ONLY,** it should require an off-set value and it must be passed like above as part of the `PRINT_START` on the same line at the end of it, not before and not after or it will give `must home first error` if set before or it will dig the sheet with the nozzle if done after and the sheet is higher than 0.000

If I have 3 klipper printers, I won't need to manually change their macros, all of them will understand `SET_GCODE_OFFSET Z_ADJUST=-0.105 MOVE=1` sent from OrcaSlicer and will immediately use the new off-set 

-----

**Comment by kylek29** [comment]
This would be beneficial to BambuLab owners as well because from what I can tell, that dropdown just sets the curr_bed_type value and the Machine Start G-Code does an if statement to set the Z-offset.

Certain 3rd party bed plates are thicker and/or have a transparent layer that mess with my X1C (Whambam). Be nice to have a "Whambam" dropdown.

**Comment by R4toLoko** [comment]
> This would be beneficial to BambuLab owners as well because from what I can tell, that dropdown just sets the curr_bed_type value and the Machine Start G-Code does an if statement to set the Z-offset.
> 
> Certain 3rd party bed plates are thicker and/or have a transparent layer that mess with my X1C (Whambam). Be nice to have a "Whambam" dropdown.

Awesome, the more printer benefiting from this, the more likely to be implemented.

Thank for the input.

**Comment by dJOS1475** [comment]
I would very much like this feature too.

**Comment by SoftFever** [comment]
I intentionally disabled it because I personally find it not very useful and prone to errors. 
There have been many instances when I forgot to set the correct bed type, which resulted in the wrong bed temperature being set

**Comment by dJOS1475** [comment]
> I intentionally disabled it because I personally find it not very useful and prone to errors. There have been many instances when I forgot to set the correct bed type, which resulted in the wrong bed temperature being set

My use case is simply to set z offset for textured vs smooth. Because I use a Super Pinda inductive probe, I need to lower the offset by -0.275 when I switch to the powder coated plate. I’d love to automate this instead of remembering to apply my presets in Klipper.

**Comment by R4toLoko** [comment]
> I intentionally disabled it because I personally find it not very useful and prone to errors. There have been many instances when I forgot to set the correct bed type, which resulted in the wrong bed temperature being set

That bed temperature is restricted exclusively to BambuLab printers so disabling it because of a single printer, is automatically punishing everybody else who doesn't own BambuLab printers :)

Like @dJOS1475 , my use case like many folks is explicitly for z-offset.

- LDO Textured: -1.1
- PEO Smooth:  -0.160
- PEO Textured: -0.18
- PEY Smooth: 0.15
- PEI Smooth: 0.04 
- PEI Textured: -0.220

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/87802b3d-e357-4376-bb4c-8b619ec3ccd7)

Because I cannot use the bed type, I have to have a ton of Printer profile for each sheet which I have forgotten to change it in the past and well, the sheet got a kiss from the nozzle.

The only way I found to automate this because hell no changing the offset manually everytime, each printer profile has this vars with its offset:

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/c9946736-2c22-48ee-bd85-4a99b6b86c7b)

macros.cfg

```
[gcode_macro PRINT_START]

    {% set z_offset = params.ZOFFSET|default(0.000)|float %}

    {% if z_offset != 0 %}
     SET_GCODE_OFFSET Z_ADJUST={ z_offset } MOVE=1; SET Z-OFFSET RIGHT AWAY
    {% endif %}
```

Klipper doesn't clean offset automatically.
![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/bf494e42-07c2-47ca-b51a-68f0d4bd69fe)

That way I "just" need to remember to pick the right printer profile with the right sheet name, and klipper takes care of the rest.
If we could use the bed type, perhaps the management would be easier with a single printer profile and bed sheets.

-----

We all know that the profiles are locked, any filament and speed profile I have is locked to this printer. 
If I get a 2.4 tomorrow, I will need to create another 10 printer profiles.

Luckily, I just "saved as" so I can still access the filament and speed profiles no matter the printer profile I get.
That is also a locker that should be removed, it adds non-necessary overhead to manage profiles.
I have tried to change that in the past by editing the JSON files but this locker shouldn't exist and the change has to be done on the slicer itself.

Take PrusaSlicer as example, the sheets profiles are managed by the printer firmware so non-Prusa printers have to either make manual changes or do what I did above. Prusa is an Apple so don't expect that to change.
With OrcaSlicer, the feature is right there, locked haha

But it does not matter if I am using the stock printer profile or my profile, I can still access the filament and speed profiles as you would expect.
IMO we have a 2-in-1 problem.

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/b060bbd6-e5a9-415c-9cc3-3b3bcf13ab25)

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/4d010b60-c3fa-41d1-97d4-30c9ea004dd0)

**Comment by HakunMatat4** [comment]
@igiannakas @Noisyfox do you think this is a hard implementation or does our pov makes sense of why this restriction should be lifted??

Thank you

**Comment by amdavidson** [comment]
I too would like to have this option so that my macros can make adjustments based on bed type. 

I'm not sure I agree with the argument that we can accidentally choose the wrong one as that's basically true of all slicer settings. I would guess that there are not many among us who have not accidentally sliced something as PLA when they meant to choose ABS or some other accidental toggle. 

If you really insist on protecting users from themselves, maybe hide access under a checkbox in settings or in the printer config.

**Comment by Neiizo** [comment]
I would also really like to see this option ! The difference in z offset between texture and non texture pei plate is non negligeable.

**Comment by Gridzilla** [comment]
Please add this back! Would really love to use custom plates for my Voron.

**Comment by SoftFever** [comment]
I have heard your voices.
I will add an option to enable multiple bed types.

**Comment by SoftFever** [comment]
Solved in 1.9.0

**Comment by Gridzilla** [comment]
Broken again in 2.0 beta

https://github.com/SoftFever/OrcaSlicer/commit/eb5a1bc7135462fb7cd04fa0853d08459516a75d

![image](https://github.com/SoftFever/OrcaSlicer/assets/4488015/806edbce-139c-421e-b86a-bfcdddff75d1)

**Comment by dJOS1475** [comment]
Howdy, I'm on 2.0.0-dev and I can't figure out how to configure each bed type on my Voron based printer - am I missing something?
